synapseclient/models/mixins/asynchronous_job.py

```diff
@@ -3,13 +3,57 @@
 import time
 from dataclasses import dataclass
 from enum import Enum
-from typing import TYPE_CHECKING, Any, Dict, Optional
+from typing import Any, Dict, Optional, Self
 
+from synapseclient import Synapse
 from synapseclient.core.constants.concrete_types import AGENT_CHAT_REQUEST
 from synapseclient.core.exceptions import SynapseError, SynapseTimeoutError
 
-if TYPE_CHECKING:
-    from synapseclient import Synapse
+ASYNC_JOB_URIS = {
+    AGENT_CHAT_REQUEST: "/agent/chat/async",
+}
+
+
+class AsynchronousCommunicator:
+    """Mixin to handle communication with the Synapse Asynchronous Job service."""
+
+    def to_synapse_request(self) -> None:
+        """Converts the request to a request expected of the Synapse REST API."""
+        raise NotImplementedError("to_synapse_request must be implemented.")
+
+    def fill_from_dict(self, synapse_response: Dict[str, str]) -> Self:
+        """
+        Converts a response from the REST API into this dataclass.
+
+        Arguments:
+            synapse_response: The response from the REST API.
+
+        Returns:
+            An instance of this class.
+        """
+        raise NotImplementedError("fill_from_dict must be implemented.")
```

Comment:
These would just be blank right? [Example](https://github.com/Sage-Bionetworks-Workflows/py-orca/blob/09bf1109b65c22960c1582ba96756796aaaa0d71/src/orca/services/base/client_factory.py#L40) from py-orca
```suggestion
```

Reply:
I think it may be either filled in or blank. If it has this `NotImplementedError` I believe if you used the mixin and didn't provide an implementation it'll raise the exception.

In the other case where you leave it blank I assume that the function would execute and return None without an error.

Reply:
Ah I didn't realize that. In that case having the error is better for sure.